ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -117,4 +119,42 @@ private static void throwOnUnknownProperty(
           + String.join(",", onlyInProvided));
     }
   }
+
+  public Optional<Character> getValueDelimiter() {
+    final String providedValueDelimiter = getString(CommonCreateConfigs.VALUE_DELIMITER_PROPERTY);
+    if (providedValueDelimiter != null) {
+      if (providedValueDelimiter.isEmpty()) {
+        throw new KsqlException("Error in WITH clause property '"
+            + CommonCreateConfigs.VALUE_DELIMITER_PROPERTY
+            + "': Delimiter cannot be empty or whitespace."
+            + System.lineSeparator()
+            + "For tab or space delimited use 'TAB' or 'SPACE' as the delimeter."
+        );
+      } else if (providedValueDelimiter.length() == 1) {
+        return Optional.of(providedValueDelimiter.charAt(0));
+      } else {
+        final Character delim = NAMED_DELIMITERS.get(providedValueDelimiter);
+        if (delim != null) {
+          return Optional.of(delim);
+        } else {
+          throw new KsqlException("Error in WITH clause property '"
+              + CommonCreateConfigs.VALUE_DELIMITER_PROPERTY
+              + "': Delimiter must be a single character, 'TAB' or 'SPACE'."
+              + System.lineSeparator()
+              + "Example valid value: ';'"
+          );
+        }
+      }
+    } else {
+      return Optional.empty();
+    }
+  }
+
```

Comment:
Can we add a method that uses the keys from `NAMED_DELIMITERS` to enumerate named delimiters for error message strings? Makes it easier to maintain if we add more.

Reply:
Hi Rohan, could you elaborate a bit on what you're thinking of here?